server.py

```diff
@@ -127,6 +137,12 @@ def get_verified_email(token):
     if not re.match('.+@.+', sub):  # <-- TODO: Use a proper parser.
         return {'error': 'Invalid email: %s' % sub}
 
+    # Make sure the nonce cannot be used on further attempts.
+    nonce = NONCES[sub]
```

Comment:
I'm missing code that will handle the case where the email address is not in `NONCES`, I think?